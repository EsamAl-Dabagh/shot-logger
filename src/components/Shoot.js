import React, { Component } from 'react';

class Shoot extends Component {
  render() {
    return (
      <React.Fragment>
        <h3> This is the Shoot component </h3>
        <button className="new-take-btn">+</button>
      </React.Fragment>
    )
  }
}

export default Shoot;