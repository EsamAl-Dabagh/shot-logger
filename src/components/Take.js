import React, { Component } from 'react';
import TextField from '@material-ui/core/TextField';
import Switch from '@material-ui/core/Switch';

class Take extends Component {
  constructor() {
    super();

    this.state = {
      scene: "",
      shot: "",
      takeNumber: "",
      description: "",
      goodTake: false,
      notes: ""
    }
  }

  render() {
    return (
      <div className="take">
        <p>This is the TAKE component: {this.props.id}</p>
        <form>
          <TextField 
            className="scene-input" 
            variant="outlined" 
            label="Scene"
            autoFocus
            onChange={event => this.setState({ scene: event.target.value })} 
          />
          <TextField 
            className="shot-input"
            variant="outlined"
            label="Shot"
            onChange={event => this.setState({ shot: event.target.value })} 
          />
          <TextField 
            className="take-input"
            variant="outlined" 
            label="Take"
            onChange={event => this.setState({ takeNumber: event.target.value })} 
          />
          <TextField 
            className="desc-input"
            variant="outlined" 
            label="Description"
            onChange={event => this.setState({ description: event.target.value })} 
          />
              
          <Switch 
            className="goodTake-switch"
            checked={this.state.goodTake}
            onChange={event => this.setState({ goodTake: event.target.checked })} 
            value="goodTake"
            color="primary"
            label="Good Take?"
          />
          
          <TextField
            className="notes-input"
            variant="outlined"
            label="Notes"
            multiline
            onChange={event => this.setState({ notes: event.target.value })}
          />
        </form>
      </div>
      
    )
  }
}

export default Take;