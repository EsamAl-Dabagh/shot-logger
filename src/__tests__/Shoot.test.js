import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Shoot from '../components/Shoot';
import Take from '../components/Take';

describe("Shoot", () => {

  const shoot = shallow(<Shoot />);

  afterEach( () => {
    shoot.setState({
      takes: []
    })
  })

  it("renders correctly", () => {
    expect(shoot).toMatchSnapshot();
  })

  it("initializes the `state` with an empty list of takes", () => {
    expect(shoot.state().takes).toEqual([]);
  })

  describe("add new take button", () => {

    it("adds a new take to `state` when add button is clicked", () => {
      shoot.find(".new-take-btn").simulate("click");

      expect(shoot.state().takes.length).toEqual(1)

    });

    it("assigns a unique id to each take", () => {
      shoot.find(".new-take-btn").simulate("click");
      shoot.find(".new-take-btn").simulate("click");
      shoot.find(".new-take-btn").simulate("click");
      
      expect(shoot.state().takes[0].props.id).toEqual(1);
      expect(shoot.state().takes[1].props.id).toEqual(2);
      expect(shoot.state().takes[2].props.id).toEqual(3);
    });
    
    it("adds a new Take row to the 'takes' div", () => {
      shoot.find(".new-take-btn").simulate("click");
      const displayedTakes = shoot.find(".takes");

      expect(displayedTakes.children().length).toEqual(1);

    });

  });
  
});