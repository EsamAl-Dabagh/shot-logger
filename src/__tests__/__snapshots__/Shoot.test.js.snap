// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoot renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Shoot />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ShootTitle />,
        <WithStyles(Button)
          className="new-take-btn"
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
          Add New Take
        </WithStyles(Button)>,
        <div
          className="takes"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Add New Take",
          "className": "new-take-btn",
          "color": "primary",
          "onClick": [Function],
          "variant": "contained",
        },
        "ref": null,
        "rendered": "Add New Take",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "takes",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ShootTitle />,
          <WithStyles(Button)
            className="new-take-btn"
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            Add New Take
          </WithStyles(Button)>,
          <div
            className="takes"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add New Take",
            "className": "new-take-btn",
            "color": "primary",
            "onClick": [Function],
            "variant": "contained",
          },
          "ref": null,
          "rendered": "Add New Take",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "takes",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
